source/Ship.cpp

```diff
@@ -4057,3 +4099,29 @@ void Ship::CreateSparks(vector<Visual> &visuals, const Effect *effect, double am
 			visuals.emplace_back(*effect, angle.Rotate(point) + position, velocity, angle);
 	}
 }
+
+
+
+double Ship::CalculateAttraction() const
+{
+	return max(0., .4 * sqrt(attributes.Get("cargo space")) - 1.8);
+}
+
+
+
+double Ship::CalculateDeterrence() const
+{
+	double tempDeterrence = 0.;
+	for(const Hardpoint &hardpoint : Weapons())
+		if(hardpoint.GetOutfit())
+		{
+			const Outfit *weapon = hardpoint.GetOutfit();
+			if(weapon->Ammo() && !OutfitCount(weapon->Ammo()))
+				continue;
```

Comment:
While we're here, this should also check AmmoUsage so that it also considers weapons that don't consume ammo.